source/common/grpc/async_client_manager_impl.cc

```diff
@@ -138,5 +141,38 @@ AsyncClientManagerImpl::factoryForGrpcService(const envoy::config::core::v3::Grp
   return nullptr;
 }
 
+RawAsyncClientSharedPtr AsyncClientManagerImpl::getOrCreateRawAsyncClient(
+    const envoy::config::core::v3::GrpcService& config, Stats::Scope& scope,
+    bool skip_cluster_check, bool always_use_cache) {
```

Comment:
per discussion can we use an enum for always_use_cache? I think that'll make the call-sites more self-documenting than another bool.